code/game/atoms.dm

```diff
@@ -679,6 +679,21 @@ its easier to just keep the beam vertical.
 		var/obj/O = o
 		O.show_message(message,2,deaf_message,1)
 
+/atom/movable/proc/dropInto(var/atom/destination)
+	while(istype(destination))
+		var/atom/drop_destination = destination.onDropInto(src)
+		if(!istype(drop_destination) || drop_destination == destination)
+			return forceMove(destination)
+		destination = drop_destination
+	return forceMove(null)
+
+/atom/proc/onDropInto(var/atom/movable/AM)
+	return // If onDropInto returns null, then dropInto will forceMove AM into us.
+
+/atom/movable/onDropInto(var/atom/movable/AM)
+	return loc // If onDropInto returns something, then dropInto will attempt to drop AM there.
```

Comment:
Could you explain what is this code meant to do in general ? it looks like a really fancy way to get the turf of a atom..

Reply:
not sure, I think it's for the dropInto code to work since it didn't exist on our code already and no matter how long I stared at the other procs used for moving stuff I couldn't figure out how to plug anything else into it